<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The major goal of ggtranscript is the simplify the process of 
    visualizing transcript structure and annotation. ggtranscript achieves by 
    introducing 4 geoms (geom_range, geom_half_range, geom_intron and
    geom_junction) and several helper functions that are designed to visualize 
    transcript annotation using ggplot2. As a ggplot2 extension, ggtranscript 
    inherits ggplot2's familiarity and flexibility; users can intuitively adjust 
    aesthetics and parameters as well as complement ggtranscript geoms with 
    existing ggplot2 geoms. ">
<title>Visualising transcript structure and annotation using ggplot2 • ggtranscript</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Visualising transcript structure and annotation using ggplot2">
<meta property="og:description" content="The major goal of ggtranscript is the simplify the process of 
    visualizing transcript structure and annotation. ggtranscript achieves by 
    introducing 4 geoms (geom_range, geom_half_range, geom_intron and
    geom_junction) and several helper functions that are designed to visualize 
    transcript annotation using ggplot2. As a ggplot2 extension, ggtranscript 
    inherits ggplot2's familiarity and flexibility; users can intuitively adjust 
    aesthetics and parameters as well as complement ggtranscript geoms with 
    existing ggplot2 geoms. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">ggtranscript</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/ggtranscript.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/dzhang32/ggtranscript/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="ggtranscript-">ggtranscript <img src="reference/figures/ggtranscript_logo_cropped.svg" align="right" height="139"><a class="anchor" aria-label="anchor" href="#ggtranscript-"></a>
</h1></div>
<!-- badges: start -->

<p><code>ggtranscript</code> is a <code>ggplot2</code> extension that makes it easy visualize transcript structure and annotation.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"dzhang32/ggtranscript"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p><code>ggtranscript</code> introduces 3 new <code>geom</code>s designed to visualise transcript annotation; <code><a href="reference/geom_range.html">geom_range()</a></code>, <code><a href="reference/geom_intron.html">geom_intron()</a></code> and <code><a href="reference/geom_junction.html">geom_junction()</a></code>.</p>
<p><code><a href="reference/geom_range.html">geom_range()</a></code> and <code><a href="reference/geom_intron.html">geom_intron()</a></code> enable the plotting of exons and introns. <code>ggtranscript</code> also provides a useful helpful function <code>to_intron</code> to convert exon co-ordinates to the corresponding introns.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dzhang32/ggtranscript" class="external-link">ggtranscript</a></span><span class="op">)</span>

<span class="co"># gene annotation for the an example gene (GBA)</span>
<span class="va">gba_ens_105</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 × 8</span>
<span class="co">#&gt;   seqnames  start    end strand type  gene_name transcript_name transcript_biot…</span>
<span class="co">#&gt;   &lt;fct&gt;     &lt;int&gt;  &lt;int&gt; &lt;fct&gt;  &lt;fct&gt; &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;           </span>
<span class="co">#&gt; 1 1        1.55e8 1.55e8 -      gene  GBA       &lt;NA&gt;            &lt;NA&gt;            </span>
<span class="co">#&gt; 2 1        1.55e8 1.55e8 -      tran… GBA       GBA-202         protein_coding  </span>
<span class="co">#&gt; 3 1        1.55e8 1.55e8 -      exon  GBA       GBA-202         protein_coding  </span>
<span class="co">#&gt; 4 1        1.55e8 1.55e8 -      CDS   GBA       GBA-202         protein_coding  </span>
<span class="co">#&gt; 5 1        1.55e8 1.55e8 -      star… GBA       GBA-202         protein_coding  </span>
<span class="co">#&gt; 6 1        1.55e8 1.55e8 -      exon  GBA       GBA-202         protein_coding</span>

<span class="co"># obtain exons</span>
<span class="va">gba_ens_105_exons</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span>

<span class="va">gba_ens_105_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">transcript_biotype</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">gba_ens_105_exons</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-tx-annot-base-1.png" width="100%"></p>
<p><code>geom_range</code> can be used for any genomic range based annotation. For example, when plotting protein-coding transcripts, it can be useful to visually distinguish the coding segments from UTRs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># keeping only the exons from protein coding transcripts</span>
<span class="va">gba_ens_105_exons_prot_cod</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span>

<span class="co"># obtain cds</span>
<span class="va">gba_ens_105_cds</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CDS"</span><span class="op">)</span>

<span class="va">gba_ens_105_exons_prot_cod</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.25</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">gba_ens_105_cds</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">gba_ens_105_exons_prot_cod</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-tx-annot-w-cds-1.png" width="100%"></p>
<p>When working with short-read RNA-sequencing data, it can be useful to check whether a known transcript structure has junction support using <code><a href="reference/geom_junction.html">geom_junction()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># using two transcripts as an example</span>
<span class="va">gba_ens_105_201_exons</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GBA-201"</span><span class="op">)</span><span class="op">)</span>

<span class="va">gba_ens_105_201_cds</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_cds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"GBA-201"</span><span class="op">)</span>

<span class="co"># simulate junction data, randomly keeping half of the junctions</span>
<span class="va">gba_ens_105_201_introns</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="reference/to_intron.html">to_intron</a></span><span class="op">(</span><span class="st">"transcript_name"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">32</span><span class="op">)</span>

<span class="va">gba_ens_105_201_junctions</span> <span class="op">&lt;-</span>
    <span class="va">gba_ens_105_201_introns</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gba_ens_105_201_introns</span><span class="op">)</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span>, <span class="op">]</span>

<span class="va">gba_ens_105_201_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        xstart <span class="op">=</span> <span class="va">start</span>,
        xend <span class="op">=</span> <span class="va">end</span>,
        y <span class="op">=</span> <span class="va">transcript_name</span>
    <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        fill <span class="op">=</span> <span class="st">"white"</span>,
        height <span class="op">=</span> <span class="fl">0.25</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_range.html">geom_range</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">gba_ens_105_201_cds</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_intron.html">geom_intron</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">gba_ens_105_201_introns</span>,
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>,
        arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span>,
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="reference/geom_junction.html">geom_junction</a></span><span class="op">(</span>
        data <span class="op">=</span> <span class="va">gba_ens_105_201_junctions</span>,
        colour <span class="op">=</span> <span class="st">"red"</span>
    <span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-tx-annot-w-junction-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>ggtranscript</code> project is released with a <a href="http://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="development-tools">Development tools<a class="anchor" aria-label="anchor" href="#development-tools"></a>
</h2>
<ul>
<li>Continuous code testing is possible thanks to <a href="https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/" class="external-link">GitHub actions</a> through <em><a href="https://CRAN.R-project.org/package=usethis" class="external-link">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes" class="external-link">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck" class="external-link">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/" class="external-link">Bioconductor’s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.14/BiocCheck" class="external-link">BiocCheck</a></em>.</li>
<li>Code coverage assessment is possible thanks to <a href="https://codecov.io/gh" class="external-link">codecov</a> and <em><a href="https://CRAN.R-project.org/package=covr" class="external-link">covr</a></em>.</li>
<li>The <a href="http://dzhang32.github.io/ggtranscript" class="external-link">documentation website</a> is automatically updated thanks to <em><a href="https://CRAN.R-project.org/package=pkgdown" class="external-link">pkgdown</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler" class="external-link">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools" class="external-link">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2" class="external-link">roxygen2</a></em>.</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.14/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/dzhang32/ggtranscript/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ggtranscript</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>David Zhang <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2382-8460" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Emil Gustavsson <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0541-7537" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/dzhang32/ggtranscript/issues" class="external-link"><img src="https://img.shields.io/github/issues/dzhang32/ggtranscript" alt="GitHub issues"></a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/dzhang32/ggtranscript" alt="GitHub pulls"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/dzhang32/ggtranscript/actions" class="external-link"><img src="https://github.com/dzhang32/ggtranscript/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check-bioc"></a></li>
<li><a href="https://app.codecov.io/gh/dzhang32/ggtranscript?branch=main" class="external-link"><img src="https://codecov.io/gh/dzhang32/ggtranscript/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
