<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot junction curves — geom_junction • ggtranscript</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot junction curves — geom_junction"><meta property="og:description" content="geom_junction() draws curves that are designed to represent junction reads
from RNA-sequencing data. The curves can be modified using curvature,
angle and ncp parameters. By default, the junctions will alternate
between being plotted on the top and bottom of each y group, however this
can be changed via junction.orientation."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggtranscript</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/ggtranscript.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/ggtranscript/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot junction curves</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/geom_junction.R" class="external-link"><code>R/geom_junction.R</code></a></small>
    <div class="hidden name"><code>geom_junction.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>geom_junction()</code> draws curves that are designed to represent junction reads
from RNA-sequencing data. The curves can be modified using <code>curvature</code>,
<code>angle</code> and <code>ncp</code> parameters. By default, the junctions will alternate
between being plotted on the top and bottom of each <code>y</code> group, however this
can be changed via <code>junction.orientation</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">geom_junction</span><span class="op">(</span>
  mapping <span class="op">=</span> <span class="cn">NULL</span>,
  data <span class="op">=</span> <span class="cn">NULL</span>,
  stat <span class="op">=</span> <span class="st">"identity"</span>,
  position <span class="op">=</span> <span class="st">"identity"</span>,
  <span class="va">...</span>,
  junction.orientation <span class="op">=</span> <span class="st">"alternating"</span>,
  curvature <span class="op">=</span> <span class="fl">0.5</span>,
  angle <span class="op">=</span> <span class="fl">90</span>,
  ncp <span class="op">=</span> <span class="fl">5</span>,
  arrow <span class="op">=</span> <span class="cn">NULL</span>,
  arrow.fill <span class="op">=</span> <span class="cn">NULL</span>,
  lineend <span class="op">=</span> <span class="st">"butt"</span>,
  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,
  show.legend <span class="op">=</span> <span class="cn">NA</span>,
  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mapping</dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code> or
<code><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_()</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), it is combined with the default mapping at the top level of the
plot. You must supply <code>mapping</code> if there is no plot mapping.</p></dd>
<dt>data</dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>
<dt>stat</dt>
<dd><p>The statistical transformation to use on the data for this
layer, as a string.</p></dd>
<dt>position</dt>
<dd><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p></dd>
<dt>...</dt>
<dd><p>Other arguments passed on to <code><a href="https://ggplot2.tidyverse.org/reference/layer.html" class="external-link">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p></dd>
<dt>junction.orientation</dt>
<dd><p><code>character</code> one of "alternating", "top" or
"bottom". Specifies where the junctions will be plotted with respect to
each <code>y</code> group.</p></dd>
<dt>curvature</dt>
<dd><p>A numeric value giving the amount of curvature.
    Negative values produce left-hand curves, positive values
    produce right-hand curves, and zero produces a straight line.</p></dd>
<dt>angle</dt>
<dd><p>A numeric value between 0 and 180,
    giving an amount to skew the control
    points of the curve.  Values less than 90 skew the curve towards
    the start point and values greater than 90 skew the curve
    towards the end point.</p></dd>
<dt>ncp</dt>
<dd><p>The number of control points used to draw the curve.
    More control points creates a smoother curve.</p></dd>
<dt>arrow</dt>
<dd><p>specification for arrow heads, as created by arrow().</p></dd>
<dt>arrow.fill</dt>
<dd><p>fill colour to use for the arrow head (if closed). <code>NULL</code>
means use <code>colour</code> aesthetic.</p></dd>
<dt>lineend</dt>
<dd><p>Line end style (round, butt, square).</p></dd>
<dt>na.rm</dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>
<dt>show.legend</dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>
<dt>inherit.aes</dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">example_introns</span> <span class="op">&lt;-</span></span>
<span class="r-in">  <span class="va">gba_ens_105</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span>group_var <span class="op">=</span> <span class="va">transcript_name</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">example_introns</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 77 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames strand type   gene_name transcript_name transcript_biotype     start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 1        -      intron GBA       GBA-202         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 1        -      intron GBA       GBA-201         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 1        -      intron GBA       GBA-203         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 1        -      intron GBA       GBA-204         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 1        -      intron GBA       GBA-211         processed_transcript  1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 1        -      intron GBA       GBA-210         processed_transcript  1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 1        -      intron GBA       GBA-206         processed_transcript  1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 1        -      intron GBA       GBA-202         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 1        -      intron GBA       GBA-201         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 1        -      intron GBA       GBA-203         protein_coding        1.55<span style="color: #949494;">e</span>8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 67 more rows, and 1 more variable: end &lt;dbl&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base</span> <span class="op">&lt;-</span> <span class="va">example_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"GBA-202"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">  <span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span>junction.orientation <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span class="va">base</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span>junction.orientation <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-3.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># for multiple transcripts, sometimes the junctions will overlap</span></span>
<span class="r-in"><span class="va">base_multi_transcript</span> <span class="op">&lt;-</span> <span class="va">example_introns</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span class="r-in">    xstart <span class="op">=</span> <span class="va">start</span>,</span>
<span class="r-in">    xend <span class="op">=</span> <span class="va">end</span>,</span>
<span class="r-in">    y <span class="op">=</span> <span class="va">transcript_name</span></span>
<span class="r-in">  <span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base_multi_transcript</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-4.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># this can be corrected using the curvature parameter</span></span>
<span class="r-in"><span class="va">base_multi_transcript</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span>curvature <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-5.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">base_multi_transcript</span> <span class="op">+</span> <span class="fu">geom_junction</span><span class="op">(</span></span>
<span class="r-in">  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">transcript_name</span><span class="op">)</span>,</span>
<span class="r-in">  curvature <span class="op">=</span> <span class="fl">0.25</span></span>
<span class="r-in">  <span class="op">)</span></span>
<span class="r-plt img"><img src="geom_junction-6.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

