<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Improve transcript structure visualisation by shortening intron gaps — shorten_gaps • ggtranscript</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Improve transcript structure visualisation by shortening intron gaps — shorten_gaps"><meta property="og:description" content="shorten_gaps is a helper function intended to improve visualizaing
transcript structure when a transcript has long introns. shorten_gaps does
this by shortening gaps (regions that do not overlap any exons) to a
user-inputted target_gap_width, then rescaling introns and exons to
preserve exon alignment. shorten_gaps will never modify the size of input
exons. Importantly, the outputted rescaled co-ordinates should only be used
for visualization as they will not match the original genomic coordinates."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggtranscript</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/ggtranscript.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/ggtranscript/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Improve transcript structure visualisation by shortening intron gaps</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/ggtranscript/blob/HEAD/R/shorten_gaps.R" class="external-link"><code>R/shorten_gaps.R</code></a></small>
    <div class="hidden name"><code>shorten_gaps.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>shorten_gaps</code> is a helper function intended to improve visualizaing
transcript structure when a transcript has long introns. <code>shorten_gaps</code> does
this by shortening gaps (regions that do not overlap any <code>exons</code>) to a
user-inputted <code>target_gap_width</code>, then rescaling <code>introns</code> and <code>exons</code> to
preserve exon alignment. <code>shorten_gaps</code> will never modify the size of input
<code>exons</code>. Importantly, the outputted rescaled co-ordinates should only be used
for visualization as they will not match the original genomic coordinates.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">shorten_gaps</span><span class="op">(</span><span class="va">exons</span>, <span class="va">introns</span>, group_var <span class="op">=</span> <span class="cn">NULL</span>, target_gap_width <span class="op">=</span> <span class="fl">100L</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>exons</dt>
<dd><p><code>data.frame</code> containing exonic ranges, which can contain data
from from multiple transcripts.</p></dd>
<dt>introns</dt>
<dd><p><code>data.frame</code> containing the corresponding introns to <code>exons</code>.
This can be created by applying <code><a href="to_intron.html">to_intron()</a></code> to the <code>exons</code>. If not using
<code><a href="to_intron.html">to_intron()</a></code>, make sure intron start/end are defined as those precisely
matching the adjacent exon boundaries, rather than exon end + 1 and exon
start - 1.</p></dd>
<dt>group_var</dt>
<dd><p><code>character</code> if input <code>data.frame</code> contains more than 1
transcript, <code>group_var</code> should specify the column that differentiates
transcripts (e.g. "transcript_id").</p></dd>
<dt>target_gap_width</dt>
<dd><p><code>integer</code> specifying the size in base pairs to
shorten the gaps to.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <code>data.frame</code> that contains the co-ordinates of introns (with
shortened gaps) and exons of each input transcript, rescaled to keep exons
aligned.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">gba_ens_105_exons</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># first, let's use an example of a single transcript</span></span>
<span class="r-in"><span class="va">single_tx</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GBA-203"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">single_tx</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 10 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames     start       end strand type  gene_name transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 1        155<span style="text-decoration: underline;">241</span>086 155<span style="text-decoration: underline;">241</span>275 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 1        155<span style="text-decoration: underline;">240</span>630 155<span style="text-decoration: underline;">240</span>717 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 1        155<span style="text-decoration: underline;">239</span>886 155<span style="text-decoration: underline;">240</span>077 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 1        155<span style="text-decoration: underline;">238</span>517 155<span style="text-decoration: underline;">238</span>650 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 1        155<span style="text-decoration: underline;">238</span>134 155<span style="text-decoration: underline;">238</span>306 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 1        155<span style="text-decoration: underline;">237</span>341 155<span style="text-decoration: underline;">237</span>578 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 1        155<span style="text-decoration: underline;">236</span>245 155<span style="text-decoration: underline;">236</span>469 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 1        155<span style="text-decoration: underline;">235</span>681 155<span style="text-decoration: underline;">235</span>844 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 1        155<span style="text-decoration: underline;">235</span>195 155<span style="text-decoration: underline;">235</span>311 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 1        155<span style="text-decoration: underline;">234</span>559 155<span style="text-decoration: underline;">235</span>100 -      exon  GBA       GBA-203        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 1 more variable: transcript_biotype &lt;chr&gt;</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">single_tx_rescaled</span> <span class="op">&lt;-</span> <span class="fu">shorten_gaps</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">single_tx</span>,</span>
<span class="r-in">    <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">single_tx</span><span class="op">)</span>,</span>
<span class="r-in">    group_var <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span class="r-in">    target_gap_width <span class="op">=</span> <span class="fl">100L</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># rescaled output contains both introns and exons</span></span>
<span class="r-in"><span class="va">single_tx_rescaled</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 19 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames start   end strand type   gene_name transcript_name transcript_biot…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> 1            2   543 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> 1          543   638 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> 1          638   754 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> 1          754   855 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> 1          855  <span style="text-decoration: underline;">1</span>018 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> 1         <span style="text-decoration: underline;">1</span>018  <span style="text-decoration: underline;">1</span>119 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> 1         <span style="text-decoration: underline;">1</span>119  <span style="text-decoration: underline;">1</span>343 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> 1         <span style="text-decoration: underline;">1</span>343  <span style="text-decoration: underline;">1</span>444 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> 1         <span style="text-decoration: underline;">1</span>444  <span style="text-decoration: underline;">1</span>681 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> 1         <span style="text-decoration: underline;">1</span>681  <span style="text-decoration: underline;">1</span>782 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> 1         <span style="text-decoration: underline;">1</span>782  <span style="text-decoration: underline;">1</span>954 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> 1         <span style="text-decoration: underline;">1</span>954  <span style="text-decoration: underline;">2</span>055 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> 1         <span style="text-decoration: underline;">2</span>055  <span style="text-decoration: underline;">2</span>188 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> 1         <span style="text-decoration: underline;">2</span>188  <span style="text-decoration: underline;">2</span>289 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> 1         <span style="text-decoration: underline;">2</span>289  <span style="text-decoration: underline;">2</span>480 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> 1         <span style="text-decoration: underline;">2</span>480  <span style="text-decoration: underline;">2</span>581 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span> 1         <span style="text-decoration: underline;">2</span>581  <span style="text-decoration: underline;">2</span>668 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">18</span> 1         <span style="text-decoration: underline;">2</span>668  <span style="text-decoration: underline;">2</span>769 -      intron GBA       GBA-203         protein_coding  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">19</span> 1         <span style="text-decoration: underline;">2</span>769  <span style="text-decoration: underline;">2</span>958 -      exon   GBA       GBA-203         protein_coding  </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># create function to compare transcript visualisation</span></span>
<span class="r-in"><span class="co"># before and after shortening gaps</span></span>
<span class="r-in"><span class="va">plot_before_after</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">before_shorten_gaps</span>,</span>
<span class="r-in">                              <span class="va">after_shorten_gaps</span>,</span>
<span class="r-in">                              <span class="va">group_var</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">before_plot</span> <span class="op">&lt;-</span> <span class="va">before_shorten_gaps</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">        <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_string</a></span><span class="op">(</span></span>
<span class="r-in">            xstart <span class="op">=</span> <span class="st">"start"</span>,</span>
<span class="r-in">            xend <span class="op">=</span> <span class="st">"end"</span>,</span>
<span class="r-in">            y <span class="op">=</span> <span class="va">group_var</span></span>
<span class="r-in">        <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">        <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">        <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">            data <span class="op">=</span> <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">before_shorten_gaps</span>, <span class="va">group_var</span><span class="op">)</span>,</span>
<span class="r-in">            strand <span class="op">=</span> <span class="st">"-"</span>,</span>
<span class="r-in">            arrow.min.intron.length <span class="op">=</span> <span class="fl">500</span></span>
<span class="r-in">        <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="va">after_plot</span> <span class="op">&lt;-</span> <span class="va">after_shorten_gaps</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">        <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_.html" class="external-link">aes_string</a></span><span class="op">(</span></span>
<span class="r-in">            xstart <span class="op">=</span> <span class="st">"start"</span>,</span>
<span class="r-in">            xend <span class="op">=</span> <span class="st">"end"</span>,</span>
<span class="r-in">            y <span class="op">=</span> <span class="va">group_var</span></span>
<span class="r-in">        <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">        <span class="fu"><a href="geom_range.html">geom_range</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span class="r-in">        <span class="fu"><a href="geom_intron.html">geom_intron</a></span><span class="op">(</span></span>
<span class="r-in">            data <span class="op">=</span> <span class="va">after_shorten_gaps</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">                <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"intron"</span><span class="op">)</span>,</span>
<span class="r-in">            strand <span class="op">=</span> <span class="st">"-"</span>,</span>
<span class="r-in">            arrow.min.intron.length <span class="op">=</span> <span class="fl">200</span></span>
<span class="r-in">        <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="va">before_after_plotlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">before_plot</span>, <span class="va">after_plot</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="va">before_after_plot</span> <span class="op">&lt;-</span> <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span></span>
<span class="r-in">        plotlist <span class="op">=</span> <span class="va">before_after_plotlist</span>,</span>
<span class="r-in">        nrow <span class="op">=</span> <span class="fl">2</span>,</span>
<span class="r-in">        labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Before shorten_gaps()"</span>, <span class="st">"After shorten_gaps()"</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">before_after_plot</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">plot_before_after</span><span class="op">(</span></span>
<span class="r-in">    before_shorten_gaps <span class="op">=</span> <span class="va">single_tx</span>,</span>
<span class="r-in">    after_shorten_gaps <span class="op">=</span> <span class="va">single_tx_rescaled</span>,</span>
<span class="r-in">    group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-plt img"><img src="shorten_gaps-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># exons can also contain multiple transcripts</span></span>
<span class="r-in"><span class="va">multi_tx</span> <span class="op">&lt;-</span> <span class="va">gba_ens_105_exons</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span class="r-in">        <span class="va">transcript_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"GBA-20"</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">multi_tx_rescaled</span> <span class="op">&lt;-</span> <span class="fu">shorten_gaps</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">multi_tx</span>,</span>
<span class="r-in">    <span class="fu"><a href="to_intron.html">to_intron</a></span><span class="op">(</span><span class="va">multi_tx</span>, <span class="st">"transcript_name"</span><span class="op">)</span>,</span>
<span class="r-in">    group_var <span class="op">=</span> <span class="st">"transcript_name"</span>,</span>
<span class="r-in">    target_gap_width <span class="op">=</span> <span class="fl">100L</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">plot_before_after</span><span class="op">(</span></span>
<span class="r-in">    before_shorten_gaps <span class="op">=</span> <span class="va">multi_tx</span>,</span>
<span class="r-in">    after_shorten_gaps <span class="op">=</span> <span class="va">multi_tx_rescaled</span>,</span>
<span class="r-in">    group_var <span class="op">=</span> <span class="st">"transcript_name"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-plt img"><img src="shorten_gaps-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang, Emil Gustavsson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

